docs/CNDP_README.md

```diff
@@ -0,0 +1,323 @@
+<!--
+SPDX-License-Identifier: Apache-2.0
+Copyright 2020 Intel Corporation
+-->
+
+# **Cloud Native Data Plane (CNDP)**
+
+Cloud Native Data Plane (CNDP) is a collection of user space libraries for accelerating packet processing for cloud applications. It aims to provide better performance than that of standard network socket interfaces by using an I/O layer primarily built on AF_XDP, an interface that delivers packets directly to user space, bypassing the kernel networking stack. For more details refer https://cndp.io/
+
+CNDP BESS port enables sending/receiving packets to/from network interface using AF-XDP.  CNDP integration with OMEC BESS UPF enables a software-based datapath and also provides deployment flexibility for kubernetes based deployments where CNDP uses [AF-XDP device plugin](https://github.com/intel/afxdp-plugins-for-kubernetes).
+
+Following are the steps required to build and test CNDP BESS UPF docker image:
+
+### Step 1: Build the OMEC UPF docker container
+
+> Note: If you are behind a proxy make sure to export/setenv http_proxy and https_proxy
+
+From the top level directory call:
+
+```
+$ make docker-build
+```
+
+### Step 2: Test setup
+
+Following diagram shows is a test setup. 
+
+![Test setup] (./images/cndp-omec-upf-test-setup.jpg)
+
+There are two systems: System 1 is runs CNDP BESS UPF and System 2 runs DPDK based packet generator which simulates traffic generated from multiple UEs and App servers. 
+
+Install NIC driver in System 1 from this link :  https://sourceforge.net/projects/e1000/files/ice%20stable/1.9.7/. This driver is used instead of in-tree kernel driver since this driver supports tc filter for creating queue groups and RSS for GTPU packet traffic steering. We need to compile and install this driver in your system with ADQ (Application Device Queues). As mentioned in the NIC driver README, use the following command to install the NIC driver.
```

Comment:
```suggestion
Install NIC driver in System 1 from this link:  https://sourceforge.net/projects/e1000/files/ice%20stable/1.9.7/. This driver is used instead of in-tree kernel driver since this driver supports tc filter for creating queue groups and RSS for GTPU packet traffic steering. We need to compile and install this driver in your system with ADQ (Application Device Queues). As mentioned in the NIC driver README, use the following command to install the NIC driver.
```